src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DefaultRazorDocumentMappingService.cs

```diff
@@ -334,6 +335,18 @@ public override bool TryMapToProjectedDocumentPosition(RazorCodeDocument codeDoc
         public override bool TryMapToProjectedDocumentOrNextCSharpPosition(RazorCodeDocument codeDocument, int absoluteIndex, [NotNullWhen(true)] out Position? projectedPosition, out int projectedIndex)
             => TryMapToProjectedDocumentPositionInternal(codeDocument, absoluteIndex, nextCSharpPositionOnFailure: true, out projectedPosition, out projectedIndex);
 
+        public override bool TryMapToVSProjectedDocumentPosition(RazorCodeDocument codeDocument, int absoluteIndex, [NotNullWhen(true)] out VSPosition? projectedPosition, out int projectedIndex)
+        {
+            if (TryMapToProjectedDocumentPositionInternal(codeDocument, absoluteIndex, nextCSharpPositionOnFailure: false, out var omniSharpProjectedPosition, out projectedIndex))
+            {
+                projectedPosition = new VSPosition(omniSharpProjectedPosition.Line, omniSharpProjectedPosition.Character);
```

Comment:
I created `omniSharpProjectedPosition.AsVSPosition()` as a convenience method, and so we could have a better idea of where we're creating a VSPosition to convert from O# (IE, where we'll eventually need to clean stuff out).

Reply:
Rebased and used your variant of this API